unlock-app/src/hooks/useDataForGitcoinPassport.ts

```diff
@@ -0,0 +1,60 @@
+import { useQuery } from '@tanstack/react-query'
+import { storage } from '~/config/storage'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+
+const getDataForGitcoinPassport = async (
+  network: number,
+  lockAddress: string,
+  recipients: string[]
+) => {
+  try {
+    const response = await storage.getDataForRecipientsAndGitcoinPassport(
+      network,
+      lockAddress,
+      recipients
+    )
+    return response.data.result
+  } catch (error: any) {
+    if (error.response) {
+      // Propagate the error as is to be caught by the calling function
+      throw error
+    }
+    // Fallback for other types of errors
+    throw new Error('An unexpected error occurred')
+  }
+}
+
+interface UseDataForGitcoinPassportProps {
+  lockAddress: string
+  network: number
+  recipients: string[]
+}
+
+export function useDataForGitcoinPassport({
+  lockAddress,
+  network,
+  recipients,
+}: UseDataForGitcoinPassportProps) {
+  return useQuery(
+    ['getDataForGitcoinPassport', lockAddress, network, recipients],
+    async () => {
+      try {
+        return await getDataForGitcoinPassport(network, lockAddress, recipients)
```

Comment:
What happens here when there are timeouts on the locksmith (due to the 3 sec wait)?

Reply:
it would throw an error that was set in locksmith. the error is a generic "Failed to submit address for scoring" message. this is shown to the client via the `ToastHelper` component